sanskrit_parser/parser/vakya_analyzer.py

```diff
@@ -0,0 +1,190 @@
+                        help="Do not modify the input/output string to match conventions", default=False)
+    parser.add_argument('--score', dest="score", action='store_true',
+                        help="Use the lexical scorer to score the splits and reorder them")
+    return parser.parse_args(argv)
+
+
+def main(argv=None):
+    global need_lakara
+    args = getArgs(argv)
+    vgraph = None
+    _console_logging()
+    logger.info(f"Input String: {args.data}")
+    need_lakara = args.need_lakara
+    if args.debug:
+        logging.basicConfig(filename='VakyaAnalyzer.log', filemode='w', level=logging.DEBUG)
+    s = LexicalSandhiAnalyzer(args.lexical_lookup)
+    if args.input_encoding is None:
+        ie = None
+    else:
+        ie = SanskritBase.SCHEMES[args.input_encoding]
+    i = SanskritBase.SanskritNormalizedString(args.data, encoding=ie,
+                                              strict_io=args.strict_io,
+                                              replace_ending_visarga=None)
+    logger.info(f"Input String in SLP1: {i.canonical()}")
+    import time
+    logger.info("Start Split")
+    start_split = time.time()
+    graph = s.getSandhiSplits(i, tag=True)
+    end_split = time.time()
+    logger.info("End DAG generation")
+    with SanskritBase.outputctx(args.strict_io):
+        if graph:
+            start_path = time.time()
+            splits = graph.find_all_paths(max_paths=args.max_paths, score=args.score)
+            end_path = time.time()
+            logger.info("End pathfinding")
+            print("Splits:")
+            for sp in splits:
+                print(f"Lexical Split: {sp}")
+                logger.info(f"Lexical Split: {sp}")
+                vgraph = VakyaGraph(sp, max_parse_dc=args.split_above)
+                for (ix, p) in enumerate(vgraph.parses):
+                    print(f"Parse {ix}")
+                    t = []
+                    for n in sorted(list(p), key=lambda x: x.index):
+                        preds = list(p.predecessors(n))
+                        if preds:
+                            pred = preds[0]  # Only one
+                            lbl = p.edges[pred, n]['label']
+                            t.append(jedge(pred, n, lbl))
+                        else:
+                            t.append(jnode(n))
+                    for e in t:
+                        if e[2]:
+                            print(f"{e[0]} => {e[1]} : {e[2]} of {e[3]}")
+                        else:
+                            print(f"{e[0]} => {e[1]}")
+
+            logger.info("End Morphological Analysis")
+            logger.info("-----------")
+            logger.info("Performance")
+            logger.info("Time taken for split: {0:0.6f}s".format(end_split-start_split))
+            logger.info(f"Time taken for path: {end_path-start_path:0.6f}s")
+        else:
+            logger.warning("No Valid Splits Found")
+    return vgraph
```

Comment:
This `vgraph` appears to correspond to the last `split` found since it is overwritten for each split on line 161. Is this the intent?

Reply:
Yes. This is not great, we need to think about this a bit more.

Reply:
Yes. And it's ugly.